java-checks/src/main/java/org/sonar/java/checks/InnerStaticClassesCheck.java

```diff
@@ -123,18 +123,18 @@ private int referenceInstance(Symbol symbol) {
       if (symbol.isUnknown()) {
         result = atLeastOneReference.size() - 1;
       } else if (!symbol.isStatic()) {
-        result = fromInstance(owner);
+        result = fromInstance(symbol, owner);
       }
     }
     return result;
   }
 
-  private int fromInstance(Symbol owner) {
+  private int fromInstance(Symbol symbol, Symbol owner) {
     int i = -1;
     Type ownerType = owner.type();
     for (Symbol outerClass : outerClasses) {
       i++;
-      if (owner.equals(outerClass) || (ownerType != null && owner.isTypeSymbol() && outerClass.type().isSubtypeOf(ownerType))) {
+      if (symbol.equals(outerClass) || owner.equals(outerClass) || (ownerType != null && owner.isTypeSymbol() && outerClass.type().isSubtypeOf(ownerType))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)